src/core/load_balancing/rls/rls.cc

```diff
@@ -1492,14 +1505,14 @@ void RlsLb::Cache::StartCleanupTimer() {
 void RlsLb::Cache::OnCleanupTimer() {
   GRPC_TRACE_LOG(rls_lb, INFO)
       << "[rlslb " << lb_policy_ << "] cache cleanup timer fired";
-  std::vector<OrphanablePtr<Entry>> entries_to_delete;
+  std::vector<RefCountedPtr<ChildPolicyWrapper>> child_policies_to_delete;
   MutexLock lock(&lb_policy_->mu_);
   if (!cleanup_timer_handle_.has_value()) return;
   if (lb_policy_->is_shutdown_) return;
   for (auto it = map_.begin(); it != map_.end();) {
     if (GPR_UNLIKELY(it->second->ShouldRemove() && it->second->CanEvict())) {
       size_ -= it->second->Size();
-      entries_to_delete.push_back(std::move(it->second));
+      child_policies_to_delete = it->second->ClearChildPolicyWrappers();
```

Comment:
I think this needs to append instead of assigning, because we might remove more than one cache entry as we iterate through the loop.

Reply:
ah, thanks, pfew!